source/features/repo-age.tsx

```diff
@@ -4,50 +4,87 @@ import cache from 'webext-storage-cache';
 import select from 'select-dom';
 import repoIcon from 'octicon/repo.svg';
 import elementReady from 'element-ready';
+import fetchDom from '../libs/fetch-dom';
 import features from '../libs/features';
-import * as api from '../libs/api';
-import {getRepoGQL, getRepoURL} from '../libs/utils';
+import {getRepoURL} from '../libs/utils';
 
 const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	year: 'numeric',
 	month: 'long',
 	day: 'numeric'
 });
 
-const getRepoCreationDate = cache.function(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository(${getRepoGQL()}) {
-			createdAt
-		}
-	`);
+type OldestCommitDetails = {
+	url: string;
+	datetime: string;
+};
 
-	return repository.createdAt;
+const getOldestCommitDetails = cache.function(async (): Promise<OldestCommitDetails | false> => {
+	await elementReady<HTMLAnchorElement>('.commit-tease-sha', {target: select('.commit-tease')!});
+	const commitsCount = Number(select('li.commits .num')!.textContent!.replace(',', ''));
+	const lastCommitHash = select<HTMLAnchorElement>('.commit-tease-sha')!.href.split('/').pop()!;
+
+	if (commitsCount === 0) {
+		return false;
+	}
+
+	if (commitsCount === 1) {
```

Comment:
I think repos with a single commit are rare enough that we can drop this condition.

Reply:
Then, What do we display for 1 commit repo?

Reply:
Without this condition, it will just continue like it would, to `fetchDom` etc.

Reply:
But the request would not return anything, the page returned by `fetchDom` would be empty and we won't find any commits there....

Reply:
Ouch. Didn’t think it would be. I guess it can stay then.